src/test/java/net/sf/jabref/importer/fileformat/BiblioscapeImporterTest.java

```diff
@@ -0,0 +1,121 @@
+package net.sf.jabref.importer.fileformat;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.importer.fileformat.BiblioscapeImporter;
+import net.sf.jabref.model.entry.BibEntry;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/*
+ * Biblioscape Tag File with an example:
+ * https://web.archive.org/web/20080820010408/http://www.biblioscape.com/manual_bsp/Biblioscape_Tag_File.htm
+ */
+
+public class BiblioscapeImporterTest {
+
+    @Before
+    public void setUp() throws Exception {
+        if (Globals.prefs == null) {
+            Globals.prefs = JabRefPreferences.getInstance();
+        }
+    }
+
+    @Test
+    public void testGetFormatName() {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        Assert.assertEquals(importer.getFormatName(), "Biblioscape");
+    }
+
+    @Test
+    public void testGetCLIID() {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        Assert.assertEquals(importer.getCLIId(), "biblioscape");
+    }
+
+    @Test
+    public void testIsRecognizedFormat() throws Throwable {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        List<String> list = Arrays.asList("BiblioscapeImporterTest.txt", "CopacImporterTest1.txt",
+                "CopacImporterTest2.txt", "CopacImporterTest3.txt", "IEEEImport1.txt", "IsiImporterTest1.isi",
+                "IsiImporterTestInspec.isi", "IsiImporterTestWOS.isi", "IsiImporterTestMedline.isi",
+                "RisImporterTest1.ris");
+        for (String str : list) {
+            try (InputStream is = BiblioscapeImporterTest.class.getResourceAsStream(str)) {
+                    Assert.assertTrue(importer.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testImportEntries() throws Throwable {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        BibtexImporter bibtexImporter = new BibtexImporter();
+
+        try (InputStream bScapeStream = BiblioscapeImporter.class.getResourceAsStream("BiblioscapeImporterTest.txt");
+                InputStream bibStream = BiblioscapeImporter.class
+                        .getResourceAsStream("BiblioscapeImporterTest_comparefile.txt")) {
+            List<BibEntry> bScapeEntries = importer.importEntries(bScapeStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibtexImporter.importEntries(bibStream, new OutputPrinterToNull());
+
+            Assert.assertEquals(bScapeEntries.size(), bibEntries.size());
```

Comment:
The problem about this is that the entry sequence is not unique.
So it is not guaranteed that the right entries are going to be compared.
BibtexEntryAssert compares the list like: `list1.get(i) == list2.get(i)`.
The best method we discovered is to compare the `isbn` field for all combinations (see l. 73) to get proper objects to compare.